### PR TITLE
pb-updates-v3

### PR DESCRIPTION
**Pull request by alibazlamit**
@nacx Here is the v3 updates with the fixed two live tests that were failing, ready for review.

Thanks.

### CODE REVIEW

File: profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/ServerApiLiveTest.java [hunk 1]
```diff
@@ -69,30 +72,31 @@ public void testCreateServer() {
               .cores(1)
               .ram(1024)
               .build());
-      
+
+      assertRequestCompleted(testServer);
```
Comment (nacx) [inline profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/ServerApiLiveTest.java]: Duplicated?

File: profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/ServerApiLiveTest.java [hunk 2]
```diff
@@ -105,149 +109,166 @@ public void testUpdateServer() {
               .ram(1024 * 2)
               .cores(2)
               .build());
-      
+
+      assertRequestCompleted(updated);
```
Comment (nacx) [inline profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/ServerApiLiveTest.java]: Duplicated?

File: profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/ServerApiLiveTest.java [hunk 3]
```diff
@@ -105,149 +109,166 @@ public void testUpdateServer() {
...
       attachedVolume = serverApi().attachVolume(
               Server.Request.attachVolumeBuilder()
               .dataCenterId(testServer.dataCenterId())
               .serverId(testServer.id())
               .volumeId(volume.id())
               .build()
       );
-      
+
+      assertRequestCompleted(attachedVolume);
```
Comment (nacx) [inline profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/ServerApiLiveTest.java]: Same here, duplicate?

**Comment by alibazlamit** [comment]
@nacx removed duplicates. and squashed.

**Comment by nacx** [comment]
Live tests are all green :)
Tests run: 94, Failures: 0, Errors: 0, Skipped: 3

Pushed to master as [61e5651b](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/61e5651b). Thanks @alibazlamit!

**Comment by nacx** [comment]
BTW, I've amended the commit message before merging.

**Comment by alibazlamit** [comment]
@nacx What is the next step for  moving this provider to production?

**Comment by nacx** [comment]
Having the live tests passing consistently, which is something we can verify now that we have a periodical build. That's the most important one.

Apart from that (and it is something the soap based provider is missing) the provider should ideally:

* Honor the templateOptions.inboundPorts, and create firewall rules for the servers accordingly.
* Implement the SecurityGroupExtension, to be able to manage firewalls and firewall rules (templates and rules in ProfitBricks?).
* Implement the ImageExtension, to be able to create an image from a running server.

If those features make sense for ProfitBricks it would be great to open a JIRA issue for each one so we can properly track them.

**Comment by alibazlamit** [comment]
@nacx how are we going to proceed with this?

**Comment by nacx** [comment]
HI @alibazlamit! I've opened [JCLOUDS-1207](https://issues.apache.org/jira/browse/JCLOUDS-1207) to track the promotion of profitbricks-rest.

**Pull request closed without merging**